migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/column.rs

```diff
@@ -75,7 +76,33 @@ impl<'a> ColumnDiffer<'a> {
             return true;
         }
 
+        // JSON defaults on MySQL should be ignored.
+        if self.diffing_options.sql_family().is_mysql()
+            && (self.previous.column_type_family().is_json() || self.next.column_type_family().is_json())
+        {
+            return true;
+        }
+
         match (&self.previous.default(), &self.next.default()) {
+            // Avoid naive string comparisons for JSON defaults.
+            (
+                Some(DefaultValue::VALUE(PrismaValue::Json(prev_json))),
+                Some(DefaultValue::VALUE(PrismaValue::Json(next_json))),
+            )
+            | (
+                Some(DefaultValue::VALUE(PrismaValue::String(prev_json))),
+                Some(DefaultValue::VALUE(PrismaValue::Json(next_json))),
+            )
+            | (
+                Some(DefaultValue::VALUE(PrismaValue::Json(prev_json))),
+                Some(DefaultValue::VALUE(PrismaValue::String(next_json))),
+            ) => serde_json::from_str::<serde_json::Value>(prev_json)
+                .and_then(|prev_json| {
+                    serde_json::from_str::<serde_json::Value>(next_json).map(|next_json| (prev_json, next_json))
+                })
+                .map(|(prev_json, next_json)| prev_json == next_json)
+                .unwrap_or_else(|_| true),
```

Comment:
note to self: `.unwrap_or(true)`

Reply:
note to self: maybe extract 99-104 to a helper function: `json_defaults_match`.

Reply:
yeah this should be fixed...